processes/gops/process_info_linux.go

```diff
@@ -1,20 +1,40 @@
 package gops
 
 import (
-	"strings"
+	"fmt"
 
 	"github.com/shirou/gopsutil/process"
 )
 
-func pickName(p *process.Process) (string, error) {
-	cmdline, err := p.Cmdline()
-	if err != nil {
-		return "", err
+// Build a hash pid -> ppid
+func buildPIDHash(processInfos []*ProcessInfo) (hash map[int32]int32) {
+	hash = make(map[int32]int32)
+	for _, processInfo := range processInfos {
+		hash[processInfo.PID] = processInfo.PPID
 	}
-	// Assume that the process is a kernel process when it has no args
-	if strings.TrimSpace(cmdline) == "" {
-		return "kernel", nil
+	return
+}
+
+// Return whether the PID is of a kernel thread, based on whether it has
+// the init process (PID 1) as ancestor
+func isKernelThread(pid int32, pidHash map[int32]int32) bool {
+	if pid == 1 {
+		return false
+	}
+	ppid, ok := pidHash[pid]
+	if !ok {
+		return true
 	}
 
-	return p.Name()
+	return isKernelThread(ppid, pidHash)
+}
+
+// Name processes "kernel" if they're a kernel thread
+func postProcess(processInfos []*ProcessInfo) {
+	pidHash := buildPIDHash(processInfos)
```

Comment:
Thanks, hadn't thought about that.

Addressed this by adding a recursion limit, let me know what you think.